src/rust/src/contract.rs

```diff
@@ -378,9 +378,11 @@ fn storage_footprint_to_ledger_footprint(
             AccessType::ReadWrite => read_write.push(k.clone()),
         }
     }
+    let read_only_not_boxed: Vec<LedgerKey> = read_only.into_iter().map(|x| *x).collect();
```

Comment:
@graydon do you know of a better way to go from `Vec<Box<LedgerKey>>` to `VecM<LedgerKey>`?

Reply:
I don't see significantly better way of doing this. If you only needed references to the ledger keys you could get a Vec<&LedgerKey>, but you actually need owned values for the LedgerFootprint. If we're going to be doing this type of conversion frequently we could add a `TryFrom<Vec<Box<T>>> for VecM<T>` to `stellar_xdr`.

Reply:
Yeah I think this is the best bet for now.